.github/workflows/accept.yml

```diff
@@ -0,0 +1,186 @@
+      matrix: ${{fromJson(needs.load-fallback-group.outputs.matrix)}}
+
+    env:
+      AO_TOKEN_PROD: ${{ secrets.AO_TOKEN_PROD }}
+
+    steps:
+      - name: Checkout ${{ github.ref }}
+        uses: actions/checkout@v2
+
+      - name: Create Container ${{ matrix.image }}
+        id: create
+        run: .github/scripts/container-from-matrix-input.sh ${{ matrix.image }}
+
+      # the current version package has yet to be built.
+      # it will fallback to build from source
+      - name: NPM Install (to fallback)
+        run: docker exec ${{ steps.create.outputs.containerId }} npm install --unsafe-perm
+
+      - name: Run tests
+        run: docker exec ${{ steps.create.outputs.containerId }} npm test
+        if: ${{ github.event.inputs.target-test }}
+
+      - name: Stop & Remove container
+        if: always()
+        run: |
+          docker stop ${{ steps.create.outputs.containerId }}
+          docker rm --force ${{ steps.create.outputs.containerId }}
+
+  # entry point 2: build group, prebuilt group
+  load-build-group:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+    - name: Checkout ${{ github.ref }}
+      uses: actions/checkout@v2
+
+      # *** comments for: build-group.json
+      # build with the lowest versions of the OSes supported so the glibc/musl versions are the oldest/most compatible. 
+      # note: some of those images are no longer supported officially (https://hub.docker.com/_/node)
+    - name: Load build group data
+      id: set-matrix
+      run: .github/scripts/matrix-from-json.sh .github/config/build-group.json
+
+  build-group-publish:
+    runs-on: ubuntu-latest 
+    needs: load-build-group
+    strategy:
+      fail-fast: false
+      matrix: ${{fromJson(needs.load-build-group.outputs.matrix)}}
+
+    env:
+      AO_TOKEN_PROD: ${{ secrets.AO_TOKEN_PROD }}
+
+    steps:
+      - name: Checkout ${{ github.ref }}
+        uses: actions/checkout@v2
+
+      # *** IMPORTANT: 
+      # send staging key pair to enable container to access S3 buckets
+      - name: Create Container ${{ matrix.image }}
+        id: create
+        run: |
+         .github/scripts/container-from-matrix-input.sh \
+          ${{ matrix.image }} \
+          ${{ secrets.AWS_ACCESS_KEY_ID }} \
+          ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+
+      # must install specific dependencies before a build
+      # can't call npm install. doing so may fallback-to-build if package has yet to be published (double build)
+      # use npm workaround specifying a package name to bypass install script in package.json
+      - name: NPM Install dependencies
+        run: docker exec ${{ steps.create.outputs.containerId }} npm install linux-os-info --unsafe-perm
+
+
+      # runs: node setup-liboboe.js && node-pre-gyp install --build-from-source
+      - name: NPM Install with Rebuild from source
+        run: docker exec ${{ steps.create.outputs.containerId }} npm run rebuild
```

Comment:
Wondering why this step is needed, if the next one `npx node-pre-gyp rebuild` ends up compiling the bindings again?

Also, in the [original workflow](https://github.com/appoptics/appoptics-bindings-node/blob/master/.github/actions/generic/image-actions/publish.sh#L61-L63), the three pre-gyp commands are: `configure rebuild package`.  Did you determine that `configure` isn't really needed?

Reply:
Step was 'half-needed' ;) Replaced a full build-from-source step with an explicit call to `node setup-liboboe.js` before `node-pre-gyp rebuild`. Both .github/workflows/accept.yml and .github/workflows/release.yml

`node-pre-gyp rebuild` runs "clean" and "build" at once. It is mapped to `node-gyp rebuild` which internally means "clean + configure + build" and triggers a full recompile. (here: https://github.com/mapbox/node-pre-gyp/blob/master/lib/rebuild.js and https://github.com/mapbox/node-pre-gyp/blob/master/lib/build.js). Explicit `configure` not needed.